h2o-py/h2o/frame.py

```diff
@@ -1280,34 +1284,63 @@ def drop(self, index, axis=1):
           H2OFrame with the respective dropped columns or rows. Returns a new H2OFrame.
         """
         if axis == 1:
-            if is_type(index, str):
-                index = self.names.index(index)
+            if not isinstance(index,list):
+                #If input is a string, i.e., "C1":
+                if is_type(index, str):
+                    #Check if index is an actual column(s) in the frame
+                    assert set([index]).issubset(self.names) == True, "Column(s) selected to drop are not in " \
```

Comment:
The condition can be more succinctly written as `index in self.names`

Also asserts are best used for internal consistency checks. For testing user input it's better to check it explicitly and then raise an `H2OValueError` if it fails.